contracts/root/RootERC20Predicate.sol

```diff
@@ -71,22 +75,49 @@ contract RootERC20Predicate is Initializable, IRootERC20Predicate {
      * @inheritdoc IRootERC20Predicate
      */
     function deposit(IERC20Metadata rootToken, uint256 amount) external {
+        uint256 expectedBalance = rootToken.balanceOf(address(this)) + amount;
         _deposit(rootToken, msg.sender, amount);
+        // invariant check to ensure that the root token balance has increased by the amount deposited
+        require((rootToken.balanceOf(address(this)) == expectedBalance), "RootERC20Predicate: UNEXPECTED BALANCE");
```

Comment:
This check is required in case the token's `transfer` implementation has a fee and the full amount is not received